app/models/foreman_tasks/task.rb

```diff
@@ -158,6 +158,30 @@ def add_missing_task_groups(groups)
       (groups - task_groups).each { |group| task_groups << group }
     end
 
+    def cancelled?
+      execution_plan.errors.map(&:exception).any? { |exception| exception.class == ::ForemanTasks::Task::TaskCancelledException }
```

Comment:
I'm a bit concerned about the `cancelled` calculation: I would suggest removing this from here, and perhaps introducing different state when we want to query based on this. I suspect this being a potential reason to this BZ https://bugzilla.redhat.com/show_bug.cgi?id=1426204